#! /bin/sh /usr/share/dpatch/dpatch-run
## 06_fix_makedep.dpatch by Stefan Fritsch <sf@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Part 2 of workaround for Makedep madness

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' sam2p-0.47-1~/ccdep.pl sam2p-0.47-1/ccdep.pl
--- sam2p-0.47-1~/ccdep.pl	2008-11-15 00:08:22.000000000 +0100
+++ sam2p-0.47-1/ccdep.pl	2011-03-02 23:34:59.776000431 +0100
@@ -162,6 +162,7 @@
 my $BS; # undef
 while ($R=~/\G(.*)\n?/g) {
   my $S=$1;
+  print $R;
 
   if ($S=~/\\\Z/) { if(defined$BS){$BS.="\n$S"}else{$BS=$S} next }
   if (defined $BS) { $S="$BS\n$S"; undef $BS }
@@ -224,6 +225,9 @@
   } elsif ($S=~/: No such file or directory$/) {
     # ^^^ gcc-3.3
     undef $included_from;
+  } elsif ($S=~/compilation terminated/) {
+    # ^^^ gcc-4.5
+    undef $included_from;
   } elsif ($S=~/^distcc\[/) {
   } else {
     die "$0: invalid depret: [$S]\n";
